packages/boost-histogram/test_boost_histogram.py

```diff
@@ -0,0 +1,35 @@
+from pyodide_build.testing import run_in_pyodide
+
+
+@run_in_pyodide(packages=["boost-histogram"])
+def test_boost_histogram():
+    import boost_histogram as bh
+
+    h = bh.Histogram(bh.axis.Integer(0, 10))
+    h.fill([1, 1, 1, 14])
+    assert h[bh.underflow] == 0
+    assert h[bh.loc(0)] == 0
+    assert h[bh.loc(1)] == 3
+    assert h[bh.overflow] == 1
+
+    assert h.sum() == 3
+    assert h.sum(flow=True) == 4
+    assert h[sum] == 4
```

Comment:
Okay, that is... creative? I don't think I would have thought of this. The resulting behavior is very strange too. The number that is being thrown seems to be a memory address that is aligned to 8 bytes.

Reply:
It is *not* a `PyObject*`, it is *not* a string. It seems to point to the four bytes `[76, -81, 9, 1]`, which is `17411916`. This is itself divisible by 4 (but not 8) and a reasonable size for a pointer. Looking at memory address `17411916`, it contains  6444 which is most likely a function pointer to `std::runtime_error::~runtime_error()`.

Reply:
If that's the case, the number being thrown is a pointer to a pointer to a function pointer...

Reply:
Okay, this is the result of an uncaught C++ exception:
```js
  __cxa_throw: function(ptr, type, destructor) {
#if EXCEPTION_DEBUG
    err('Compiled code throwing an exception, ' + [ptr,type,destructor]);
#endif
    var info = new ExceptionInfo(ptr);
    // Initialize ExceptionInfo content after it was allocated in __cxa_allocate_exception.
    info.init(type, destructor);
    exceptionLast = ptr;
    uncaughtExceptionCount++;
    throw ptr;
  }
```

Reply:
The exception is being thrown inside of a function whose mangled name is 1014 bytes long:
```C++
pybind11::cpp_function::initialize<register_histogram<boost::histogram::storage_adaptor<std::__2::vector<double, std::__2::allocator<double> > > >(pybind11::module_&, char const*, char const*)::{lambda(boost::histogram::histogram<std::__2::vector<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean>, std::__2::allocator<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean> > >, boost::histogram::storage_adaptor<std::__2::vector<double, std::__2::allocator<double> > > > const&, pybind11::args&)#1}, double, boost::histogram::histogram<std::__2::vector<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean>, std::__2::allocator<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean> > >, boost::histogram::storage_adaptor<std::__2::vector<double, std::__2::allocator<double> > > > const&, pybind11::args&, pybind11::name, pybind11::is_method, pybind11::sibling>(register_histogram<boost::histogram::storage_adaptor<std::__2::vector<double, std::__2::allocator<double> > > >(pybind11::module_&, char const*, char const*)::{lambda(boost::histogram::histogram<std::__2::vector<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean>, std::__2::allocator<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean> > >, boost::histogram::storage_adaptor<std::__2::vector<double, std::__2::allocator<double> > > > const&, pybind11::args&)#1}&&, double (*)(boost::histogram::histogram<std::__2::vector<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean>, std::__2::allocator<boost::histogram::axis::variant<boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<11u> >, boost::histogram::axis::regular<double, boost::use_default, metadata_t, boost::histogram::axis::option::bitset<6u> >, boost::histogram::axis::regular<double, boost::histogram::axis::transform::pow, metadata_t, boost::use_default>, boost::histogram::axis::regular<double, func_transform, metadata_t, boost::use_default>, axis::regular_numpy, boost::histogram::axis::variable<double, metadata_t, boost::use_default, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<1u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<0u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<11u>, std::__2::allocator<double> >, boost::histogram::axis::variable<double, metadata_t, boost::histogram::axis::option::bitset<6u>, std::__2::allocator<double> >, boost::histogram::axis::integer<int, metadata_t, boost::use_default>, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<1u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<2u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<0u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<8u> >, boost::histogram::axis::integer<int, metadata_t, boost::histogram::axis::option::bitset<4u> >, boost::histogram::axis::category<int, metadata_t, boost::use_default, std::__2::allocator<int> >, boost::histogram::axis::category<int, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<int> >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<2u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, boost::histogram::axis::category<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> >, metadata_t, boost::histogram::axis::option::bitset<8u>, std::__2::allocator<std::__2::basic_string<char, std::__2::char_traits<char>, std::__2::allocator<char> > > >, axis::boolean> > >, boost::histogram::storage_adaptor<std::__2::vector<double, std::__2::allocator<double> > > > const&, pybind11::args&), pybind11::name const&, pybind11::is_method const&, pybind11::sibling const&)::{lambda(pybind11::detail::function_call&)#1}::operator()(pybind11::detail::function_call) 
```

Reply:
The exception is thrown as the last wasm instruction in:
https://github.com/pybind/pybind11/blob/a25d40c7bceb4db227aeb4a44fc74e28eb1015f1/include/pybind11/pybind11.h#L1276

Reply:
The error message is:
```
Unable to cast Python instance to C++ type (compile in debug mode for details)
```

Reply:
I wonder if something is causing `sum` to leak through into a C++ call. `[sum]` should convert to `slice(None, None, sum)` in https://github.com/scikit-hep/boost-histogram/blob/724b0b2e59f04c625d6248103596f3fdda32f2d1/src/boost_histogram/_internal/hist.py#L623 then be converted into the correct Boost.Histogram command in https://github.com/scikit-hep/boost-histogram/blob/724b0b2e59f04c625d6248103596f3fdda32f2d1/src/boost_histogram/_internal/hist.py#L811.

If I can activate and see logging, there should be logging commands before the C++ calls here. I think I can get this locally too (slightly different message, but close).

Reply:
Are these C++ exceptions supposed to be caught and raised as a `RuntimeError` in normal Python? Are they expected to occur ever or are they supposed to reflect programmer errors?

We probably can't get them to be non panicking without enabling exceptions which I think causes a performance hit (maybe an acceptable one?) but I think I can improve the resulting error message a lot.

Reply:
They are supposed to be seen from Python. If you pass Python objects into a pybind11 function that you don't have a conversion for, you get a RuntimeError.

In fact, turning off EH would break this: https://github.com/scikit-hep/boost-histogram/blob/724b0b2e59f04c625d6248103596f3fdda32f2d1/src/boost_histogram/_internal/hist.py#L765-L771

Reply:
pybind11 uses exceptions pretty heavily: https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html

Reply:
That said, this one is pretty easy to work around in Python if needed, I can just verify this really is a tuple of integers rather than just trying it and catching the C++ exception.

Reply:
Can you add a test that triggers a different one of these C++ exceptions?

Reply:
Oh sorry I didn't read your message carefully:

> For boost-histogram, I think this is the only place this occurs

Reply:
That's the only place we catch one.

Reply:
The rest of them are "real" errors, that is, they should show up to the user, and are not handled.

Reply:
Apparently wasm exception handling is pretty far along, it's been released in Chrome 96 in September and by Safari 15.2 in December. Firefox only has it on firefox nightly. According to the v8 people, enabling wasm exception handling has a negligible performance cost.

We could also just try enabling js exception handling and see how big the cost is.

Reply:
Huh, well it seems like we do have js exception handling enabled because otherwise it would just say:

> 'DISABLE_EXCEPTION_THROWING was set (likely due to -fno-exceptions), which means no C++ exception throwing support code is linked in, but such support is required by symbol ' + name + '. Either do not set DISABLE_EXCEPTION_THROWING (if you do want exception throwing) or compile all source files with -fno-except (so that no exceptions support code is required); also make sure DISABLE_EXCEPTION_CATCHING is set to the right value - if you want exceptions, it should be off, and vice versa.'